src/xmpp/iq.c

```diff
@@ -308,6 +309,21 @@ iq_disco_info_request(gchar *jid)
 }
 
 void
+iq_disco_info_request_onconnect(gchar *jid)
+{
+    xmpp_ctx_t * const ctx = connection_get_ctx();
+    char *id = create_unique_id("disco_info_onconnect");
```

Comment:
Fixed in a branch https://github.com/boothj5/profanity/tree/xep_0363_http_file_upload_pr

If you merge this to your branch I think it will update this pull request.